### PR TITLE
Extract Network State

### PR DESCRIPTION
**Pull request by shrugs**
(will be rebased off of `mino` after #152 is merged)

This follows the same pattern as the routing PR where we extract the network state from the top level component and instead of passing it down the tree component-by-component we use a context provider to provide the state and mutating functions.

We also simplify our web3 usage by deriving `web3` and `contracts` from the network type.

- rename to `NETWORK_TYPES`
- use single `compose` lib in walletgenerator
- create `online` hook/hoc for walletgenerator (and eventually bridge) to use for online status

### CODE REVIEW

File: src/walletgen/Walletgen.js [hunk 1]
```diff
@@ -19,8 +19,8 @@ import Done from './views/Done';
 
 import './style/index.css';
 
-const checkNetwork = () =>
-  navigator.onLine ? NETWORK_STATES.ONLINE : NETWORK_STATES.OFFLINE;
+// TODO: actually merge these two projects into one
```
Comment (Fang-) [inline src/walletgen/Walletgen.js]: Not entirely sure what's meant by this.

File: src/bridge/lib/lib.js [hunk 2]
```diff
@@ -1,26 +1,26 @@
 import * as ob from 'urbit-ob';
 
-const compose = (...fs) =>
+export const compose = (...fs) =>
```
[the remark this answers is not in the document]
Comment (shrugs) [reply src/bridge/lib/lib.js]: just a style/practices preference, I don't think they're semantically any different

File: src/bridge/store/online.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+export const ONLINE_STATUS = {
+  ONLINE: Symbol('ONLINE'),
+  OFFLINE: Symbol('OFFLINE'),
+  UNKNOWN: Symbol('UNKNOWN'),
+};
+
+const getNetworkState = () => {
+  if ('onLine' in navigator) {
+    return navigator.onLine ? ONLINE_STATUS.ONLINE : ONLINE_STATUS.OFFLINE;
```
[the remark this answers is not in the document]
Comment (shrugs) [reply src/bridge/store/online.js]: ah, navigator is the global from the browser, and `onLine` is the name of the API: https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/onLine

File: src/walletgen/Walletgen.js (hunk 1 above)
Comment (shrugs) [reply src/walletgen/Walletgen.js]: it's referencing a file from bridge directly, which breaks the "these two projects are totally separate" assumption, but that will change once wallet generation is an actual feature of bridge

File: src/bridge/store/online.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Fang-) [reply src/bridge/store/online.js]: Oh lol, that's awkward. Well, we'll live with it. (:

File: src/bridge/lib/lib.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Fang-) [reply src/bridge/lib/lib.js]: Personally prefer all-in-one export at the end of the file. Not gonna _make_ you revert this, but just fyi. (;

**Pull request merged**